src/MelodyGenerator.js

```diff
@@ -3,18 +3,122 @@ const MelodyQueue = require('./MelodyQueue.js');
 const waveGenerator = require('./WaveGenerator.js');
 const note = require('./note.js');
 
-class MelodyGenerator
-{
+class MelodyGenerator {
     fromString(melodyString) {
         // put your code here
+        this.defines = [];
+        this.sounds = [];
+        let step1 = melodyString.replace(new RegExp(', ', "g"), ',').split("\n");
+
+        for (let i = 0; i < step1.length; i++) {
```

Comment:
I see many old for loops. Maybe foreach like `for(let step of step1)` or `step1.forEach(function)`

see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/forEach

Reply:
For loop is maybe old, but the fastest loop in js. 
See: https://hackernoon.com/javascript-performance-test-for-vs-for-each-vs-map-reduce-filter-find-32c1113f19d7

Reply:
You don't know what I need :) Here is my counterargument: https://www.phpied.com/extreme-javascript-optimization/ Especially at the bottom line...

Reply:
`You don't know what I need :) ` :fire: